scripts/installer/curl_install_pypi/install.py

```diff
@@ -0,0 +1,420 @@
+    print_status("Extracting '{}' to '{}'.".format(download_location, tmp_dir))
+    package_tar = tarfile.open(download_location)
+    package_tar.extractall(path=tmp_dir)
+    package_tar.close()
+    virtualenv_dir_name = 'virtualenv-'+VIRTUALENV_VERSION
+    working_dir = os.path.join(tmp_dir, virtualenv_dir_name)
+    cmd = [sys.executable, 'virtualenv.py', '--python', sys.executable, install_dir]
+    exec_command(cmd, cwd=working_dir)
+
+
+def install_cli(install_dir, tmp_dir):
+    path_to_pip = os.path.join(install_dir, 'bin', 'pip')
+    cmd = [path_to_pip, 'install', '--cache-dir', tmp_dir, 'promptflow[azure,executable,pfs,azureml-serving]',
+           '--upgrade']
+    exec_command(cmd)
+    cmd = [path_to_pip, 'install', '--cache-dir', tmp_dir, 'promptflow-tools', '--upgrade']
+    exec_command(cmd)
+
+
+def create_executable(exec_dir, install_dir):
+    create_dir(exec_dir)
+    exec_filepaths = []
+    for filename, template in [(PF_EXECUTABLE_NAME, PF_DISPATCH_TEMPLATE),
+                               (PFAZURE_EXECUTABLE_NAME, PFAZURE_DISPATCH_TEMPLATE),
+                               (PFS_EXECUTABLE_NAME, PFS_DISPATCH_TEMPLATE)]:
+        exec_filepath = os.path.join(exec_dir, filename)
+        with open(exec_filepath, 'w') as exec_file:
+            exec_file.write(template.format(install_dir=install_dir))
+        cur_stat = os.stat(exec_filepath)
+        os.chmod(exec_filepath, cur_stat.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+        print_status("The executable is available at '{}'.".format(exec_filepath))
+        exec_filepaths.append(exec_filepath)
+    return exec_filepaths
+
+
+def get_install_dir():
+    install_dir = None
+    while not install_dir:
+        prompt_message = 'In what directory would you like to place the install?'
+        install_dir = prompt_input_with_default(prompt_message, DEFAULT_INSTALL_DIR)
+        install_dir = os.path.realpath(os.path.expanduser(install_dir))
+        if ' ' in install_dir:
+            print_status("The install directory '{}' cannot contain spaces.".format(install_dir))
+            install_dir = None
+        else:
+            create_dir(install_dir)
+            if os.listdir(install_dir):
+                print_status("'{}' is not empty and may contain a previous installation.".format(install_dir))
+                ans_yes = prompt_y_n('Remove this directory?', 'n')
+                if ans_yes:
+                    shutil.rmtree(install_dir)
+                    print_status("Deleted '{}'.".format(install_dir))
+                    create_dir(install_dir)
+                else:
+                    # User opted to not delete the directory so ask for install directory again
+                    install_dir = None
+    print_status("We will install at '{}'.".format(install_dir))
+    return install_dir
+
+
+def get_exec_dir():
+    exec_dir = None
+    while not exec_dir:
+        prompt_message = (f"In what directory would you like to place the "
```

Comment:
do we follow any other tools install experience here?

Reply:
this scripts refer to az installlation

Reply:
https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=script